src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -291,31 +291,71 @@ private IEnumerable<CompletionItem> GetModulePathCompletions(SemanticModel model
-                    file.Segments.Last().EndsWith(LanguageConstants.LanguageId) ? CompletionPriority.High : CompletionPriority.Medium)
-                .Build())
-                .ToList();
 
+            // Prioritize .bicep files higher than other files.
+            var bicepFileItems = CreateFileCompletionItems(files, cwdUri, IsBicepFile, CompletionPriority.High);
+            var armTemplateFileItems = CreateFileCompletionItems(files, cwdUri, IsArmTemplateFileLike, CompletionPriority.Medium);
             var dirItems = dirs
-                .Select(dir => CreateModulePathCompletionBuilder(
-                    dir.Segments.Last(),
-                    (entered.StartsWith("./") ? "./" : "") + cwdUri.MakeRelativeUri(dir).ToString(),
-                    context.ReplacementRange,
-                    CompletionItemKind.Folder,
-                    CompletionPriority.Medium)
-                .WithCommand(new Command { Name = EditorCommands.RequestCompletions })
-                .Build())
-                .ToList();
-            return fileItems.Concat(dirItems);
+                .Select(dir =>
+                    CreateModulePathCompletionBuilder(
+                        dir.Segments.Last(),
+                        // "./" will not be preserved when making relative Uris. We have to go and manually add it.
+                        (entered.StartsWith("./") ? "./" : "") + cwdUri.MakeRelativeUri(dir).ToString(),
+                        context.ReplacementRange,
+                        CompletionItemKind.Folder,
+                        CompletionPriority.Low)
+                    .WithCommand(new Command { Name = EditorCommands.RequestCompletions })
+                    .Build());
+
+            return bicepFileItems.Concat(armTemplateFileItems).Concat(dirItems);
+
+            // Local functions.
+            IEnumerable<CompletionItem> CreateFileCompletionItems(IEnumerable<Uri> fileUris, Uri cwdUri, Predicate<Uri> predicate, CompletionPriority priority) => fileUris
+                .Where(fileUri => fileUri != model.SourceFile.FileUri && predicate(fileUri))
+                .Select(fileUri =>
+                    CreateModulePathCompletionBuilder(
+                        fileUri.Segments.Last(),
+                        (entered.StartsWith("./") ? "./" : "") + cwdUri.MakeRelativeUri(fileUri).ToString(),
+                        context.ReplacementRange,
+                        CompletionItemKind.File,
+                        priority)
+                    .Build());
+
+            bool IsBicepFile(Uri fileUri)
+            {
+                if (PathHelper.HasBicepExtension(fileUri))
+                {
+                    return true;
+                }
+
+                return model.Compilation.SourceFileGrouping.SourceFiles.Any(sourceFile =>
+                    sourceFile is BicepFile &&
+                    sourceFile.FileUri.LocalPath.Equals(fileUri.LocalPath, PathHelper.PathComparison));
```

Comment:
Are you trying to account for non-`.bicep` files which are already part of the compilation? Personally, I think I'd prefer to just keep it purely extension-based, or the completion experience could be confusing. Some examples:
1. I've got 2 non-`.bicep` files, only one referenced in my code. It may be unclear why only one shows up in completions.
2. File completions may be influenced by whether or not there's a syntax error in the code.

Reply:
Good pt. We'd better make the user experience more consistent. I changed it to check `.bicep` extension only.